node/overseer/overseer-gen/proc-macro/src/impl_builder.rs

```diff
@@ -107,46 +61,379 @@ pub(crate) fn impl_builder(info: &OverseerInfo) -> proc_macro2::TokenStream {
 		})
 		.collect::<Vec<_>>();
 
-	let generics = quote! {
-		< S, #( #baggage_generic_ty, )* >
-	};
-	let where_clause = quote! {
+	// A helper structure to carry fields of a builder to produce setters
+	struct BuilderFieldHelper {
+		// Used to distinguish baggage and subsystems
+		is_subsystem: bool,
+		// Valid for baggage as all subsystems are generics
+		is_generic: bool,
+		// Field name
+		name: Ident,
+		// Field type (either generic or not)
+		field_ty: TokenStream,
```

Comment:
Please _never_ use `TokenStream` in a struct, when there is a better way of representing that as a type.

Reply:
Yes, that was caused just because I wanted to unify baggage and subsystems. I will think about a better approach.